mesonbuild/coredata.py

```diff
@@ -1001,9 +1001,10 @@ def init_option(self, name: str, value: T.Optional[T.Any], prefix: str) -> _U:
         return self.opt_type(self.description, **keywords)
 
     def _argparse_action(self) -> T.Optional[str]:
-        if self.default is True:
-            return 'store_false'
-        elif self.default is False:
+        # If the type is a boolean, the presence of the argument in --foo form
+        # is to enable it. Disabling happens by using -Dfoo=false, which is
+        # parsed under `args.projectoptions` and does not hit this codepath.
+        if isinstance(self.default, bool):
```

Comment:
While this is obviously the right thing to do, it also means --debug, --errorlogs and --stdsplit are useless since they are the default. They are still useful in their -D form, so I don't think it's an issue.

Reply:
They're not entirely useless: they can still override `default_options:` in `project()`. But yes.